js/core/utils/shadow-dom.js

```diff
@@ -0,0 +1,59 @@
+let constructedStyleSheet = null;
+
+function createConstructedStyleSheet(rootNode) {
+    try {
+        // eslint-disable-next-line no-undef
+        return new CSSStyleSheet();
+    } catch(err) {
+        const styleEl = rootNode.ownerDocument.createElement('style');
+
+        rootNode.appendChild(styleEl);
+
+        return styleEl.sheet;
+    }
+}
+
+function processRules(styleSheets, needApplyAllStyles) {
+    [...styleSheets].forEach(sheet => {
+        // NOTE: need try/catch block for not-supported cross-domain css
+        try {
+            for(let j = 0; j < sheet.cssRules.length; j++) {
+                insertRule(sheet.cssRules[j], needApplyAllStyles);
+            }
+        } catch(err) {
+        }
+    });
+}
+
+function insertRule(rule, needApplyAllStyles) {
+    const isDxRule = needApplyAllStyles ||
+                     rule.selectorText?.includes('.dx-') ||
+                     rule.cssRules?.[0]?.selectorText?.includes('.dx-') ||
+                     rule.name?.startsWith('dx-') ||
```

Comment:
I think this will not have effect. The biggest percent of rules are covered with `rule.selectorText?.includes('.dx-')`.
If I remember correctly the `name` rule is applied only for a few rules like `@keyframes` or smth like this.

Reply:
oh, ok